src/c++/perf_analyzer/inference_profiler.h

```diff
@@ -234,6 +234,11 @@ class InferenceProfiler {
         current_value += step;
       } while (((current_value <= end) || (end == static_cast<T>(NO_LIMIT))) &&
                (meets_threshold));
+      // If there was only one concurrency we swept over and it did not meet the
+      // stability threshold, we should return an error.
+      if (current_value == (start + step) && meets_threshold == false) {
```

Comment:
Why don't we just have Profile return err instead of special casing here?
As far as I can tell all of the post-run errors in Profile() still fall through to return Success, which doesn't make sense to me.

Reply:
I think the reason is that we want to distinguish between the model returning an error and failing to stabilize or meeting the latency threshold. If we fail to meet the latency threshold, we will adjust it (at least in the binary search case).

Reply:
I think this is a symptom of not enough code trying to do too much. A bigger refactor would make this so much more readable but not really feasible in our time constraints.

Reply:
Also for the linear search case, we still want to allow for example, concurrency 1: good, concurrency 2: good, concurrency 3: not stable, now we print the error message deep inside, but we don't wanna return an error here because concurrency 1 and 2 were still valid and should have their results returned to user via csv